core/src/main/java/io/questdb/griffin/engine/groupby/SampleByFillNoneRecordCursorFactory.java

```diff
@@ -112,7 +113,13 @@ public boolean recordCursorSupportsRandomAccess() {
     }
 
     @NotNull
-    protected RecordCursor initFunctionsAndCursor(SqlExecutionContext executionContext, RecordCursor baseCursor) {
+    private RecordCursor initFunctionsAndCursor(SqlExecutionContext executionContext, RecordCursor baseCursor) {
+        long maxInMemoryRows = executionContext.getCairoSecurityContext().getMaxInMemoryRows();
+        if (maxInMemoryRows >= 0 && baseCursor.size() >= 0 && baseCursor.size() > maxInMemoryRows) {
```

Comment:
This is what we were talking about this morning. 99% of the flow goes over mispredicted branch.

https://dzone.com/articles/branch-prediction-in-java

Additionally, I would suggest that "max" value is set to Long.MAX_VALUE. With this choice of "allow all" value branching will look like:

```java
final long cursorSize = baseCursor.size();
if (cursorSize < maxInMemoryRows) {
// do the good stuff
} else {
      throw ....
}
```

Reply:
we have more naughty places that trigger branch issues, but we should stop their proliferation